### PR TITLE
Wintertodt

### PR DESCRIPTION
**Pull request by OTRD5k**
Adds a plugin that highlights tiles that are about to damage players.

Highlights tiles for both normal snow attacks

snowattack

and brazier explosions

brazier attack

Additionally, highlights the safespots that will prevent players from being hit by the normal snow attack.

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/wintertodt/WintertodtConfig.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+@ConfigGroup(
+		keyName = "wintertodt",
+		name = "Wintertodt",
+		description = "Configuration for the Wintertodt plugin"
+)
+
+public interface WintertodtConfig extends Config
+{
+	@ConfigItem(
+			position = 0,
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/plugins/wintertodt/WintertodtConfig.java]: The indent here is too large. See https://github.com/runelite/runelite/wiki/Code-Conventions for an XML file that will stop intellij from doing this.

File: runelite-client/src/main/java/net/runelite/client/plugins/wintertodt/WintertodtConfig.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+			name = "Safespot Marker",
+			description = "Whether or not snow attack safespots are shown"
+	)
+	default boolean showSafespot()
+	{
+		return true;
+	}
+
+	@ConfigItem(
+			position = 1,
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/plugins/wintertodt/WintertodtConfig.java]: Indent

File: runelite-client/src/main/java/net/runelite/client/plugins/wintertodt/WintertodtPlugin.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+import net.runelite.api.coords.WorldPoint;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.ui.overlay.Overlay;
+import java.util.Arrays;
+import java.util.Collection;
+
+@PluginDescriptor(
+		name = "Wintertodt"
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/plugins/wintertodt/WintertodtPlugin.java]: Indent

File: runelite-client/src/main/java/net/runelite/client/plugins/wintertodt/WintertodtPlugin.java [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+	private Client client;
+
+	@Inject
+	private WintertodtConfig config;
+
+	private static final int WINTERTODT_REGION = 6462;
+
+	@Getter
+	private static final WorldPoint[] safespots = {
+			new WorldPoint(1622, 3988, 0),
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/plugins/wintertodt/WintertodtPlugin.java]: Indent

File: runelite-client/src/main/java/net/runelite/client/plugins/wintertodt/WintertodtSnowOverlay.java [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+	private final WintertodtPlugin plugin;
+	private final WintertodtConfig config;
+
+	private static final int MAX_DISTANCE = 2350; //Grabbed from the cannon plugin, magic number to keep overlay from rendering in unloaded tiles
+	private static final int SNOW_ATTACK_CENTER_ID = ObjectID.NULL_502; //Id of the graphics object at center of snow attack
+	private static final int SNOW_PARTICLES_ID = GraphicID.SNOW_PARTICLES_ID; //Id of the particles that spawn around an attack that was spawned on a player
+
+
+	Brazier[] Braziers = {
+			//south east brazier
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/plugins/wintertodt/WintertodtSnowOverlay.java]: Indent

File: runelite-client/src/main/java/net/runelite/client/plugins/wintertodt/WintertodtSnowOverlay.java [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+			GraphicsObject object = graphicsObjects.get(i);
+			if (object.getId() == SNOW_PARTICLES_ID)
+			{
+				WorldPoint particleLocation = WorldPoint.fromLocal(client, object.getLocation());
+				int particleX = particleLocation.getX();
+				int particleY = particleLocation.getY();
+
+				//particles spawn on the tiles diagonal to the main attack object
+				if (particleX == x - 1 && particleY == y - 1 ||
+						particleX == x - 1 && particleY == y + 1 ||
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/plugins/wintertodt/WintertodtSnowOverlay.java]: Indent

**Comment by Reasel** [comment]
I tested this out last night on stream. It works very well. I did notice that the damage area was not marked right away. As in I could see the snow particles before the danger zone was marked.

Some things I would like to see added: 
- Customization for the color of the tile highlights.
- Some sort of notification for when you are interrupted by a hit. This is a bit trickier since you don't get interrupted when you are cutting logs, but do when you are fletching or burning. Just something I think would be really nice.
- An infobox that displays your current expected loot number. IE if you have 500 points it will show 2, if you have 750 it will show 2.5

**Comment by Reasel** [comment]
Another thing I would love to have is a way to be notified when I take damage.

**Comment by OTRD5k** [comment]
@MESLewis No idea what you want me to change. 

They are tabs, so shouldn't they be read as tab characters no matter how the editor is setup? It's not like I have it setup up to use spaces instead of tabs. Importing that XML file does nothing to change anything.

@Reasel 
- Yeah, colors would be nice. I'll add that in soon.
- This would be redundant, just use the idle notifier plugin already in runelite
- Maybe, but I don't know if that would really be useful to many people.

**Comment by MESLewis** [comment]
You have one two many tabs in the indents I marked

**Comment by OTRD5k** [comment]
@MESLewis Ahh, I see it now. Thanks, Ill fix them soon.

**Pull request closed without merging**